### **Caro leitor(ra) seja muito bem vindo(a), vamos explorar essa base comigo !?**
### `Importando Bibliotecas e Carregando o Dataset`
---
Seguiremos de forma bem detalhada uma análise de dados exploratória (**EDA**), buscaremos por insights nessa jornada!

No código abaixo importa-se as bibliotecas, carrega-se o dataset e exibe-se as primeiras 5 linhas :

Com o Data frame já carregado e exibido, consegue-se vizualisar previamente os dados a serem tratados e manipulados, na sequência há um passo muito importante, discernir melhor as colunas!

In [10]:
# Importar bibliotecas e definir aliases:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Carregar o dataset na variável 'df':
df = pd.read_csv("netflix_data.csv", header=0)

# Visualizar 5 primeiras linhas :
df.head(50)

show_id     type                                  title  \
0       s2    Movie                                   7:19   
1       s3    Movie                                  23:59   
2       s4    Movie                                      9   
3       s5    Movie                                     21   
4       s6  TV Show                                     46   
5       s7    Movie                                    122   
6       s8    Movie                                    187   
7       s9    Movie                                    706   
8      s10    Movie                                   1920   
9      s11    Movie                                   1922   
10     s14    Movie                                  2,215   
11     s15    Movie                                   3022   
12     s16    Movie                                 Oct-01   
13     s18    Movie                                 22-Jul   
14     s19    Movie                                 15-Aug   
15     s21    Movie                   ​​Kuch Bheege Alfaaz   
16     s22    Movie                           ​Goli Soda 2   
17     s23    Movie                     ​Maj Rati ​​Keteki   
18     s24    Movie                            ​Mayurakshi   
19     s28    Movie                                 #Alive   
20     s29    Movie          #AnneFrank - Parallel Stories   
21     s32    Movie                      #FriendButMarried   
22     s33    Movie                    #FriendButMarried 2   
23     s34    Movie                           #realityhigh   
24     s35    Movie                                  #Roxy   
25     s37    Movie                                #Selfie   
26     s38    Movie                             #Selfie 69   
27     s40    Movie                         ¡Ay, mi madre!   
28     s41    Movie                            Çarsi Pazar   
29     s42    Movie                             Ég man þig   
30     s43    Movie            Çok Filim Hareketler Bunlar   
31     s44    Movie                                Òlòtūré   
32     s45    Movie                               Æon Flux   
33     s47    Movie                       1 Chance 2 Dance   
34     s48    Movie                          1 Mile to You   
35     s49    Movie                    10 Days in Sun City   
36     s50    Movie                         10 jours en or   
37     s51    Movie                            10,000 B.C.   
38     s55    Movie                             100 Meters   
39     s58    Movie                             100% Halal   
40     s60    Movie                        1000 Rupee Note   
41     s61    Movie                           12 ROUND GUN   
42     s63    Movie                             13 Cameras   
43     s66    Movie                                13 Sins   
44     s69    Movie                              14 Blades   
45     s70    Movie                             14 Cameras   
46     s72    Movie                              16 Blocks   
47     s73    Movie                               17 Again   
48     s74    Movie                            18 Presents   
49     s75    Movie  1898: Our Last Men in the Philippines   

                       director  \
0             Jorge Michel Grau   
1                  Gilbert Chan   
2                   Shane Acker   
3                Robert Luketic   
4                   Serdar Akar   
5               Yasir Al Yasiri   
6                Kevin Reynolds   
7                 Shravan Kumar   
8                  Vikram Bhatt   
9                  Zak Hilditch   
10          Nottapon Boonprakob   
11                   John Suits   
12               Kunle Afolayan   
13              Paul Greengrass   
14           Swapnaneel Jayakar   
15                         Onir   
16                 Vijay Milton   
17            Santwana Bardoloi   
18                  Atanu Ghosh   
19                       Cho Il   
20  Sabina Fedeli, Anna Migotto   
21                Rako Prijanto   
22                Rako Prijanto   


###  `Entendimento Inicial dos Dados`
---

Compreender a estrutura e o conteúdo do *DataFrame* é uma etapa essencial para uma análise exploratória eficaz. Ter uma boa visão de negócio é uma habilidade fundamental para Engenheiros, Cientistas e Analistas de dados mas, é de suma importância lembrar que **viés cognitivo** pode influenciar interpretações. 

Por isso, é crucial deixar que os próprios dados guiem as decisões analíticas. Toda EDA deve ser construída com base em evidências extraídas diretamente do dataset, evitando suposições não fundamentadas.
A partir deste ponto, será feita uma investigação detalhada sobre colunas, tipos de dados, valores nulos, estatísticas básicas e de possíveis padrões/tendências em nichos de mercado da NETFLIX.



### `Estrutura do Data Frame`


In [7]:
# Dimensão do Data Frame
df.shape

(4812, 11)

4812 linhas e 11 colunas
### `Conteúdo das Colunas do Data Frame`

__________________________________________________________________________________________________________________________________________________

In [ ]:
# Vizualizar nomes das colunas:
df.columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'duration', 'description', 'genre'],
      dtype='object')

Traduzindo :

```
show_id:      Um identificador único para cada filme ou série.
type:         Indica se é um filme ou série. 
title:        O título do filme ou série. 
director:     O diretor do filme ou série. 
cast:         Os atores ou atrizes que participaram. 
country:      O país onde o filme ou série foi produzido.
date_added:   Quando o filme ou série foi adicionado à plataforma. 
release_year: O ano de lançamento do filme ou série. 
duration:     A duração do filme ou o número de temporadas de uma série. (Observação: o valor desta coluna depende do valor da coluna type)
description:  Uma breve descrição ou sinopse do filme ou série. 
genre:        O gênero do filme ou série (ex: ação, comédia, drama).
```

### `Tipos de Dados do Dataset`
---

Verificando tipos de dados, informações gerais e se o Data Frame contém valores nulos:

In [ ]:
# Verificar tipos de dados e informações gerais do Data Frame
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4812 entries, 0 to 4811
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       4812 non-null   object
 1   type          4812 non-null   object
 2   title         4812 non-null   object
 3   director      4812 non-null   object
 4   cast          4812 non-null   object
 5   country       4812 non-null   object
 6   date_added    4812 non-null   object
 7   release_year  4812 non-null   int64 
 8   duration      4812 non-null   int64 
 9   description   4812 non-null   object
 10  genre         4812 non-null   object
dtypes: int64(2), object(9)
memory usage: 413.7+ KB


- Conforme a saída, duas colunas são do tipo inteiro ```int64``` e 9 colunas são do tipo ```object```, o Data frame não contém valores nulos, Isso simplifica a parte de limpeza dos dados.

- As colunas ```release_year (Ano de Lançamento)``` e ```duration (Duração)``` já estão como ```int64 (números inteiros)```, o que é apropriado.

- A coluna ```data_added``` está como ```object``` o que para o **pandas** significa "texto"```(string)```. Para fazer análises temporais(como por exemplo: "quantos filmes foram adicionados por ano?"), precisamos converter essa coluna para um formato de data ```(datatime)```.


### `Convertendo tipos de dados, Tratando e limpado (Data Cleaning)`
---

Realizando a primeira tarefa de pré-processamento de dados, converte-se no código abaixo a coluna ```data_added``` para ```datatime``` e é checado se o código funcionou utilizando a mesma função anterior ```.info()```


Mas em primeira instância o código :

---

```df['date_added'] = pd.to_datetime(df['date_added'])```

Retornou: ***```_________________ValueError_______________```***

---

***E o motivo: havia "sujeira" nos dados na coluna ```date_added``` foi encontrado um " " (espaço em branco) antes da data ```" August 4, 2017"```
O Pandas olhou as primeiras datas do arquivo deduziu que o formato era "Nome do Mês, Dia, Ano" e tentou aplicar essa regra a todas as outras. Quando ele chegou na linha 161 a regra quebrou por causa do espaço extra no começo por isso a saída ValueError.***

---
Próxima tarefa remover os **espaços** extras no início e fim das datas da coluna, para posteriormente fazer a conversão:


In [ ]:
# Removendo espaços antes e depois da coluna
df['date_added'] = df['date_added'].str.strip()

# Converter a coluna data_added para datatime
df['date_added'] = pd.to_datetime(df['date_added'])

# Ver o tipo de dado da coluna
print(df['date_added'].dtype)

datetime64[ns]


Conversão concluída com Sucesso, e checada ! Isso significa que a fase de Limpeza de Dados está praticamente concluída!

Próximo passo verificar se ainda há valores **Nulos** no Data Frame:

In [ ]:
# Verificando a situação dos valores nulos após a limpeza da coluna 'date_added'
df.isnull().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
duration        0
description     0
genre           0
dtype: int64

Ótimo o Data frame não contém mais valores **Nulos**.

### Clusterizando os Dados

movies = df[df['type'] == 'Movie'].copy()
series = df[df['type'] == 'TV Show'].copy()

print(f"Número de Filmes: {len(filmes)}") 
print(f"Número de Séries: {len(series)}") ```



